### PR TITLE
Show score and setup firebase

### PR DESCRIPTION
**Pull request by MattAgn**
image
image

### CODE REVIEW

File: lib/screens/game.dart [hunk 1]
```diff
@@ -24,6 +26,7 @@ class Game extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final gameSystem = Provider.of<GameSystem>(context);
```
Comment (antoinefBam) [inline lib/screens/game.dart]: Maybe you can add the `listen : false` in this case ?

File: lib/ecs/systems/main.dart [hunk 2]
```diff
@@ -113,6 +113,8 @@ class GameSystem extends System {
     return walls;
   }
 
+  int get score => this.snake?.body?.length ?? 0;
```
Comment (antoinefBam) [inline lib/ecs/systems/main.dart]: This is where Dart is concise

File: lib/screens/login.dart [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      email: 'mat.alingrin+android@gmail.com',
+      password: 'papapa',
+    ))
+        .user;
+    print("over");
+    return user;
+  }
+
+  signup() {
+    print("toto");
```
Comment (antoinefBam) [inline lib/screens/login.dart]: you forgot a print

File: lib/screens/login.dart [hunk 4]
```diff
@@ -1,49 +1,129 @@
...
+  String _firebaseError;
+  bool _isSignupLoading = false;
+  bool _isLoginLoading = false;
 
-    final FirebaseUser user =
-        (await _auth.signInWithCredential(credential)).user;
-    print("signed in " + user.displayName);
-    return user;
+  _onSubmitSignup() async {
+    if (_formKey.currentState.validate()) {
```
Comment (antoinefBam) [inline lib/screens/login.dart]: Maybe you can extract `_formKey.currentState` in a specific variable as you use it many times in this function.

File: lib/screens/game.dart (hunk 1 above)
Comment (MattAgn) [reply lib/screens/game.dart]: yes bien vu !

**Pull request merged**